templates/intro-workspaces/README.ipynb

```diff
@@ -0,0 +1,334 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python my_repo/my_app.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Understanding Ray log output\n",
+    "\n",
+    "After running `my_app.py`, you should see output of the form `(do_some_work pid=29848) Doing work [repeated 4x across cluster]`. The prefix of the log message shows the function name, PID of the worker that ran the function, and if run on a remote worker, the node IP.\n",
+    "\n",
+    "The result of the log message contains stdout and stderr from the function execution. Ray will also deduplicate repetitive logs from parallel execution of functions across the cluster."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Monitoring program execution\n",
+    "\n",
+    "Depending on the cluster size, the above script may take some time to run. Try playing around with the number of worker machines, increasing the sleep time, or the number of function calls. Use the tools overviewed below to understand how Ray parallelizes the program.\n",
+    "\n",
+    "Let's overview some of the tools available to monitor Ray program execution in Workspaces.\n",
+    "\n",
+    "**Resources Panel**\n",
+    "\n",
+    "The resources panel provides basic stats about cluster utilization, as well as an indication of which worker nodes are being used. Use the resource panel as a quick overview of cluster status before diving deeper into the Ray dashboard.\n",
+    "\n",
+    "<img src=\"assets/resources-panel-stats.png\" height=400px/>\n",
+    "\n",
+    "**Ray dashboard > Jobs**\n",
+    "\n",
+    "To see the status of an active or previously run Ray job, navigate to `Ray Dashboard > Jobs` in the UI. Here you will see an overview of job progress, logs, and the ability to drill down into individual task and actors.\n",
+    "\n",
+    "<img src=\"assets/ray-dashboard-jobs.png\" height=400px/>\n",
+    "\n",
+    "**Ray dashboard > Metrics**\n",
+    "\n",
+    "View the aggregate time-series metrics for the cluster in order to diagnose job execution efficiency. The `Ray Dashboard > Metrics` page offers metrics on Ray tasks, actors, as well as hardware resource utilization of the cluster.\n",
+    "\n",
+    "<img src=\"assets/ray-dashboard-metrics.png\" height=400px/>\n",
+    "\n",
+    "**Logs Tab**\n",
+    "\n",
+    "View and search over Ray cluster and application logs in the Logs tab.\n",
+    "\n",
+    "<img src=\"assets/logs-tab.png\" height=400px/>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Dependency Management\n",
+    "\n",
+    "In order to run code across a cluster, Ray ships code and other library dependencies to other machines in *runtime envs*. In Workspaces, the code and installed PyPI packages are automatically added to the runtime env to be used by Ray.\n",
```

Comment:
Do we want a link to "runtime env": https://docs.ray.io/en/latest/ray-core/handling-dependencies.html